plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/semantics/SQLDocumentScriptItemSyntaxContext.java

```diff
@@ -33,16 +33,19 @@ public class SQLDocumentScriptItemSyntaxContext {
     @NotNull
     private final SQLQueryModel queryModel;
     private int length;
+    private boolean hasContextBoundaryAtLength;
     private boolean isDirty = false;
 
     public SQLDocumentScriptItemSyntaxContext(
         @NotNull String originalText,
         @NotNull SQLQueryModel queryModel,
-        int length
+        int length,
+        boolean hasContextBoundaryAtLength
```

Comment:
Let's remove it from constructor and add just a setter. Our constructors and function signatures become monstrous otherwise.